### PR TITLE
(almost) finish plate metadata

### PR DESCRIPTION
**Pull request by dstansby**
So almost finishes https://github.com/BioImageTools/ome-zarr-models-py/issues/37, but I was confused about one sentence so not quite. This does most of the plate metdata according to the spec though

### CODE REVIEW

File: src/ome_zarr_models/utils.py [hunk 1]
```diff
@@ -16,6 +16,9 @@ def _unique_items_validator(values: list[T]) -> list[T]:
     return values
 
 
+_AlphaNumericConstraint = StringConstraints()(pattern="[a-zA-Z0-9]*")
```
Comment (dstansby) [inline src/ome_zarr_models/utils.py:19]: Can someone sanity check my regex?

File: src/ome_zarr_models/utils.py [hunk 2]
```diff
@@ -16,6 +16,9 @@ def _unique_items_validator(values: list[T]) -> list[T]:
     return values
 
 
+_AlphaNumericConstraint = StringConstraints(pattern="[a-zA-Z0-9]*")
```
Comment (lorenzocerrone) [inline src/ome_zarr_models/utils.py:19]: looks good

File: src/ome_zarr_models/v04/plate.py [hunk 3]
```diff
@@ -1,95 +1,126 @@
...
+                errors.append(f"well path '{path}' does not contain a single '/'")
+                continue
+
+            row, column = path.split("/")
+            if row not in row_names:
+                errors.append(f"row in well path '{path}' is not in list of rows")
+            if column not in column_names:
+                errors.append(f"column in well path '{path}' is not in list of columns")
+
+        if len(errors):
```
[the remark this answers is not in the document]
Comment (d-v-b) [reply src/ome_zarr_models/v04/plate.py:124]: my preference is for ultra explicit `len(errors) > 0`, i'm not a fan of empty lists being truthy. but it's not a blocker

File: src/ome_zarr_models/v04/plate.py [hunk 4]
```diff
@@ -1,95 +1,126 @@
...
+    name: str | None = None
+    maximumfieldcount: PositiveInt | None = Field(
+        default=None,
+        description=("Maximum number of fields of view for the acquisition"),
     )
+    description: str | None = None
 
 
-class WellInPlate(Base):
+class Well(Base):
```
[the remark this answers is not in the document]
Comment (dstansby) [reply src/ome_zarr_models/v04/plate.py]: Huh that's a pain thanks for catching. I renamed to WellInplate, but left all the other classes alone. Do you think I should change the others back to *InPlate, or leave them as is in this PR?

[the remark this answers is not in the document]
Comment (imagejan) [reply src/ome_zarr_models/v04/plate.py]: I don't like the `*InPlate` or `*InWell` nomenclature very much. How about `WellAttrs` for the model attributes, and `Well` for the zarr group representation?

[the remark this answers is not in the document]
Comment (dstansby) [reply src/ome_zarr_models/v04/plate.py]: Both of these are model attributes. I think we should keep *Attrs for the top level attributes (so there's a corresponding class that stores zarr groups/arrays *and* the *Attrs metdata), and avoid them for other classes to have a clearer API. Am open to other suggestions than "WellInPlate" for this class though.

[the remark this answers is not in the document]
Comment (lorenzocerrone) [reply src/ome_zarr_models/v04/plate.py]: One alternative would be to prepend every model with `Plate`, `PlateRow`, `PlateCoulmn`, `PlateWell`. 

But this discussion should not sto this PR, we can discuss this in a later issue.

**Review by lorenzocerrone — APPROVED** [review]

**Pull request merged**